js/Class/Basket.js

```diff
@@ -0,0 +1,152 @@
+
+
+      const inputPlus = document.createElement('input');
+      inputPlus.setAttribute('value', '+');
+      inputPlus.setAttribute('operation', 'plus');
+      inputPlus.setAttribute('type', 'button')
+      inputPlus.setAttribute('idinput', this.idWork)
+      inputPlus.classList.add('input-button');
+      inputWraper.append(inputPlus);
+      inputPlus.addEventListener('click', this.plusMinusGoods.bind(this))
+
+      const buttonDelete = document.createElement('div');
+      buttonDelete.innerHTML = `<button class="input-button" operation="del" ><span type="button" operation="del" idinput="${this.idWork}" 
+      class="material-symbols-outlined"> delete</span></button>`;
+      buttonDelete.setAttribute('operation', 'del');
+      buttonDelete.setAttribute('idinput', this.idWork)
+      inputWraper.append(buttonDelete);
+      buttonDelete.addEventListener('click', this.plusMinusGoods.bind(this))
+
+
+      const out = document.createElement('div');
+      inputWraper.append(out);
+      out.classList.add('out', this.idWork);
+      this.updateBasketAndCalculate();
+   }
+
+   updateBasketAndCalculate() {
+      let sumOrder = 0;
+      for (let key in this.basketArr) {
+         sumOrder += this.basketArr[key].price * this.basketArr[key].amount;
+      };
+      this.sumOrderTotal = sumOrder;
+
+      if (!this.deleteProductHTML) {
+         const doc = document.getElementsByClassName(this.idWork);
+         const arrElementInput = document.querySelectorAll('.input-count');
+         for (let key2 of arrElementInput) {
+            let idInput = key2.getAttribute('idinput')
+            if (idInput == this.idWork) {
+               key2.setAttribute('value', this.basketArr[this.idWork].amount);
+            }
+         }
+         doc[0].innerHTML = `${this.basketArr[this.idWork].amount * this.basketArr[this.idWork].price} грн`;
+      }
+      this.deleteProductHTML = false;
+      document.querySelector('.order-sum').innerHTML = `Сумма заказа: ${this.sumOrderTotal} грн <button class="buy">Оформить заказ`;
+      document.querySelector('.buy').onclick = this.aproveOrder.bind(this);
+   };
+
+   changeAmountGoods(event) {
+      const idInput = event.target.getAttribute('idinput');
+      const amount = event.target.value;
+      this.idWork = idInput;
+      this.basketArr[this.idWork].amount = amount;
+      this.updateBasketAndCalculate();
+   }
+
+   plusMinusGoods(event) {
+      const idInput = event.target.getAttribute('idinput');
+      const operation = event.target.getAttribute('operation');
+      this.idWork = idInput;
+      switch (operation) {
+         case 'minus': if (this.basketArr[this.idWork].amount > 1) { this.basketArr[this.idWork].amount-- };
+            break;
+         case 'plus': this.basketArr[this.idWork].amount++;
+            break;
+         case 'del':
+            delete this.basketArr[this.idWork];
+            const x = event.target.parentNode.parentNode.parentNode.parentNode;
```

Comment:
`event.target.parentNode.parentNode.parentNode.parentNode`
Is there any other solutions to get your node ?